java/ql/src/Security/CWE/CWE-352/CsrfUnprotectedRequestType.qhelp

```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE qhelp SYSTEM "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>When you set up a web server to receive a request from a client without any mechanism
+for verifying that it was intentionally sent, then it is vulnerable to attack. An attacker can
+trick a client into making an unintended request to the web server that will be treated as
+an authentic request. This can be done via a URL, image load, XMLHttpRequest, etc. and can
+result in exposure of data or unintended code execution.</p>
+</overview>
+
+<recommendation>
+<p>Make sure any requests that change application state are protected from Cross Site Request Forgery (CSRF).
+Some application frameworks provide default CSRF protection for unsafe HTTP request methods (such as
+<code>POST</code>) which may change the state of the application. Safe HTTP request methods (such as
+<code>GET</code>) should be read-only and should not be used for actions that change application state.</p>
```

Comment:
```suggestion
<code>GET</code>) should only perform read-only operations and should not be used for actions that change application state.</p>
```

Reply:
Updated in https://github.com/github/codeql/pull/18288/commits/516df3b4be4f75a68438e0a965eca6b915a8e0e9